### PR TITLE
Added ability to set --service-node-port-range

### PR DESCRIPTION
**Pull request by robinpercy**
Addresses: #3052 and #3326 

From what I can tell, `--service-node-port-range` has been a valid option forever, so I haven't worried about k8s versions.

@justinsb @chrislovecnm let me know if any changes are required.

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Thanks!

/ok-to-test

File: pkg/apis/kops/componentconfig.go [hunk 1]
```diff
@@ -185,6 +185,8 @@ type KubeAPIServerConfig struct {
 	AdmissionControl []string `json:"admissionControl,omitempty" flag:"admission-control"`
 	// ServiceClusterIPRange is the service address range
 	ServiceClusterIPRange string `json:"serviceClusterIPRange,omitempty" flag:"service-cluster-ip-range"`
+	// A port range to reserve for services with NodePort visibility.
+	ServiceNodePortRange string `json:"serviceNodePortRange,omitempty" flag:"service-node-port-range"`
```
Comment (chrislovecnm) [inline pkg/apis/kops/componentconfig.go]: Would it be good to included the expected format here? 100-3000 for instance

**Comment by chrislovecnm** [comment]
Does github auto close issues when you say addresses? I know it picks up closes and fixes. Double check that the two issues get closed when this merges please.

**Comment by robinpercy** [comment]
@chrislovecnm thanks for the feedback, I'll fix up and squash later today.

**Comment by robinpercy** [comment]
@chrislovecnm I've squashed and fixed up the comments. The failed e2e test looks like a flake to me.

**Comment by chrislovecnm** [comment]
We have been having problems with e2e ... framework is not happy

**Comment by chrislovecnm** [comment]
@justinsb how does this interact with the nodeport PR you just put in??  Do we need a follow-up PR to set the security rules correctly.

**Comment by justinsb** [comment]
/lgtm

I'll put in a follow up to use this range in the security group and assign it to you for review @robinpercy :-)  This is the kubeAPIServer section which has pretty weak guarantees.

Out of interest @robinpercy what are you setting here (and why)?  Because you do have to be a little bit careful in the range you choose to avoid collisions!

**Comment by robinpercy** [comment]
Thanks @justinsb. I'm actually not using this feature, just noticed a few other folks were looking for it - and I needed a reintroduction to the codebase 😄  @felipejfc was asking for guidance on safe settings for this value in #3052 though.

**Comment by justinsb** [comment]
That's awesome @robinpercy - thanks for this.

I opened #3379 to use this value for the new support for configuring access to NodePort services (which we really want for the e2e test suite)

**Pull request merged**